### ISSUE TITLE
What's the prior distribution of NNGP and NTK?

### ISSUE THREAD

**Issue by Kangfei** [google/neural-tangents#114]

**Comment by romanngg** [comment]
The prior distribution of an infinitely-wide, untrained neural network is generally the mean-zero Gaussian process NNGP, i.e. `f(x1), f(x2) ~ N(0, kernel_fn(x1, x2, get='nngp'))` - not sure if this answers your question, lmk!

**Comment by Kangfei** [comment]
@romanngg Thanks for your answer! That's helpful for me. 

Another question is the efficiency of training/prediction an infinite-wide neural network. I noticed training and prediction of nngp is much faster than gp, (also implemented by jax https://github.com/google/jax/blob/master/examples/gaussian_process_regression.py). Whether the time complex of nngp is different from regular gp? If not, what's the time complexity of nngp?  Here I use the gradient_descent_mse_ensemble. It seems that there is no gradient descent steps  when t=None.

**Comment by romanngg** [comment]
The complexity should be similar - I believe the difference here is that what we call "training" of NNGP and NTK refers to training of the infinitely wide Bayesian or gradient descent-trained network, which turns out to be (roughly, in case of NTK) equivalent to a single Gaussian process regression with the NNGP or NTK kernel. So it's a single GP regression call. But in your example by "training" they mean tuning the hyper-parameters of the GP kernel by maximizing the likelihood, so they do one GP regression per training step. You could do the same with NNGP or NTK (you could tune some of the parameters of the infinitely-wide network), and in this case the cost should be comparable.
